src/main/java/javaslang/collection/Map.java

```diff
@@ -42,6 +46,17 @@ default V apply(K key) {
 
     <U, W> Map<U, W> map(BiFunction<? super K, ? super V, ? extends Tuple2<? extends U, ? extends W>> mapper);
 
+    /**
+     * Performs an action on value.
+     *
+     * @param mapper A {@code Function}
+     * @throws NullPointerException if {@code mapper} is null
+     */
+    default <W> Map<K, W> mapValues(Function<V, ? extends W> mapper) {
```

Comment:
This needs to be an abstract method, also in `SortedMap`.

We need to implement it with appropriate return type in `HashMap`, `LinkedHashMap` and `TreeMap`.